docs/backing_services.md

```diff
@@ -0,0 +1,150 @@
+
+> Note: Algolia is optional for development.
+>
+> If you do not indicate an API key for Algolia, the application will simply
+> return an empty result when searching
+
+In Algolia, you can use synonyms to improve full-text search (eg: "travel <=> trip").
+
+You can define these synonyms in the [synonyms.json](/server/algolia/synonyms.json) file.
+
+Deploy these synonyms with the following command:
+
+```bash
+# Path: ./FAQ/server
+npm run algolia
+```
+
+### Environment variables
+
+#### `REACT_APP_ALGOLIA_APP_ID`
+
+_Used by: Front-end & back-end_
+
+This variable is the ID of your Algolia project. You can find it on your project interface on the [Algolia dashboard](https://www.algolia.com/dashboard). It should be a string of 10 char (digit or uppercase letter).
+
+#### `REACT_APP_ALGOLIA_API_KEY_ALL`
+
+_Used by: Back-end_
+
+This variable is an API Key with the following operation access rights: `search`, `addObject`, `deleteObject`, `editSettings`. You can create an API key on your [Algolia dashboard](https://www.algolia.com/dashboard): Go to "API Keys > All API keys"
+
+#### `REACT_APP_ALGOLIA_API_KEY_SEARCH`
+
+_Used by: Front-end_
+
+This variable is an API Key with the following operation access rights: `search`. You can create an API key on your [Algolia dashboard](https://www.algolia.com/dashboard): Go to "API Keys > All API keys"
+
+> Note: Because this key is available in the front-end, you should only give him the `search` access right for security reasons.
+
+## Auth0
+
+[Auth0](https://auth0.com/) is used for authentication. This is required dependency.
+
+### Environment variables
+
+#### `REACT_APP_AUTH0_DOMAIN`
+
+_Used by: Front-end & back-end_
+
+Its value should be the Auth0 domain against which the app should authenticate. It ends with `auth0.com` and it can be found on the [Auth0 dashboard](https://manage.auth0.com).
+
+> If you have a zenika.com email address, you may use `zenika.eu.auth0.com`.
+
+#### `REACT_APP_AUTH0_CLIENTID`
+
+_Used by: Front-end & back-end_
+
+Its value should be the id of your Auth0 client against which the app should authenticate. It looks like a long string of random chars and can be found on the [Auth0 dashboard](https://manage.auth0.com): Go to "Clients > [Your client] > Settings"
+
+> If you want to use the FAQ Auth0 client, you may use `wq8LU1f5iXQ4HWL0F6Z07QDcSMgWPd1p`
+
+## Google Cloud
+
+[Google Cloud](https://cloud.google.com/) is used to host the front-end on App Engine.
+
+> Note: Google Cloud is an optional dependency
+
+### Environment variables
+
+#### `GCLOUD_APP_ID`
+
+_Used by: Front-end_
+
+Its value should be the id of your Google Cloud application ID. It looks like a slug (eg: "my-project") and can be found on the [Google Cloud dashboard](https://console.cloud.google.com/home/dashboard).
+
+## Graphcool
+
+[Graphcool](https://www.graph.cool/) is a Graphql-as-a-Service which holds the data. This is a required backing service, but you can deploy locally during development.
```

Comment:
After this, a step-by-step guide to launch a local GraphCool instance and configure the app to use it would be best I think:

> - Launch a local GraphCool instance: `graphcool local up`
> - Deploy to it: `npm run graphcool`
> - Choose `local` when asked for a cluster (only happens the first time)
> - Choose `dev` when asked for a target (only happens the first time)
> - Choose `faq` when asked for a service name (only happens the first time)
> - After deployment the CLI will show the endpoints URIs to use to configure the front-end (see Environment variables below)